docs/redesign-specs/DB_Schema/database.md

```diff
@@ -0,0 +1,199 @@
+* name : Name of the image
+* ownerUserId : Id of the user who owns the image.
+* type : Type of image (e.g. RAW, QCOW2, etc.)
+* visibility : Access level of image (e.g. private, public, shared, etc.)
+* isSnapshot : This flag is `true` in case of snapshot and not an uploaded image
+* dataStoreId : Id of the DataStore that gives information about where this image is stored. 
+
+**NOTE:** `(name, ownerUserId) pair is unique key.`
+
+***
+
+## ProvisionedNode
+
+1. Node running an operating system from the associated clone of a parent image (aka golden image). 
+2. A user can provision multiple nodes.
+3. A clone (shallow or deep copy depending on the underlying data store's capabilities) of the parent image is created and used to provision a node.
+4. Nodes are provisioned using a provisioning engine over a network owned by the user.
+5. BMI creates an interface (physical or virtual) on the provisioning engine for the users' provisioning network.
+
+* id : Id of the provisioned node (primary key)
+* macAddress : Mac address of the provisioned node (unique key)
+* parentImageId : Id of the golden image.
+* cloneName :  Name of the cloned image (unique key)
+* nic : NIC of the provisioned node 
+* ownerUserId : Id of the user who owns the provisioned node 
+* provisionEngineVlanId: Id of the provisionEngineVlan. This column represents the `(provisioning engine, vlan interface)` pair that is used to provision the given node.
+
+***
+
+## Tag
+
+1. Restore points of an exisitng provisioned image created by the user in time.
+2. Performance of Tag creation will depend on the data stores' capabilities.
+3.  All tag must be deleted before de-provisioning. 
+
+* id : Id of the tag (primary key)
+* name : Name of the tag
+* nodeId :Id of the node of which this tag is created.
+
+***
+
+## UserAccessControlList 
+
+1. Mapping between access control list and user.
+2. A user can be assosiated to multiple access control lists.
+3. An access control list can have multiple user in it.
+
+* id: Id of the User-AceesControlList mapping (primary key)
+* ACLId : Id of the ACL.
+* userId: Id of the user.
+
+**NOTE:** `(ACLId, userId) pair is unique key.`
+
+***
+
+## ImageAccessControlList 
+
+1. Mapping between access control list and image. 
+2. An access control list can be attached to multiple images.
+3. An image can be accessed by multple access control lists.
+
+* id: Id of the Image-AccessControlList mapping (primary key)
+* ACLId: Id of the ACL. 
+* imageId : Id of the image table.
+
+**NOTE:** `(ACLId, imageId) pair is unique key.`
+
+***
+
+## External Service Table (Authention/DataStore/ProvisionEngine) 
+
+1. Authentication: A service used to authenticate the legitimacy of a user request.
+2. DataStore: A filesystem service where images are stored.
+3. ProvisionEngine: A service used to (de)-provision a node.
+
+* The three above tables have `<id>` (primary key), `<name>` (unique key), `<type>`, `<url>` (unique key), `<ip>`, and `<port>`

```

Comment:
I **strongly** disagree with this abstraction. If you think you can make an authentication driver work with that much information, we need to talk. Also why have `ip, port` and also `url`. I would just keep `url`, since that will include also the protocol of communication, which you are missing here.